Fix namedtensor build

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#21609 Fix namedtensor build**

test_cpp_extensions compiles c++ extensions. When it does this, it also
needs USE_NAMEDTENSOR=1 to match the pytorch build.

The fix is to thread USE_NAMEDTENSOR=1 to the testing environment as
well.

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 21609 gh/zou3519/43/head

Differential Revision: [D15747687](https://our.internmc.facebook.com/intern/diff/D15747687)

zou3519 committed:
Fix namedtensor build

test_cpp_extensions compiles c++ extensions. When it does this, it also
needs USE_NAMEDTENSOR=1 to match the pytorch build.

The fix is to thread USE_NAMEDTENSOR=1 to the testing environment as
well.

Test Plan
- [namedtensor ci]

ezyang reviewed (approved):
Oof, this isn't the safest way to do it, but since this is temporary I'll let it slide.

zou3519 commented:
@ezyang what were you thinking of for how to make this safer?

ezyang commented:
Basically, make it harder to forget setting the environment variable.
The gold standard would be to write the macro into Config.h at configure
time and then use that (instead of passing it via build system.)

I don't think you /have/ to do this (since we'll kill this code when
the prototype is done) but it is safer.

Excerpts from Richard Zou's message of 2019-06-10 14:22:33 -0700:
> @ezyang what were you thinking of for how to make this safer?
>

zou3519 committed:
Update on "Fix namedtensor build"

Fix namedtensor build

test_cpp_extensions compiles c++ extensions. When it does this, it also
needs USE_NAMEDTENSOR=1 to match the pytorch build.

The fix is to thread USE_NAMEDTENSOR=1 to the testing environment as
well.

Test Plan
- [namedtensor ci]

gh-metadata: pytorch pytorch 21609 gh/zou3519/43/head
